articles/quickstart/webapp/aspnet-core-beta/01-login.md

```diff
@@ -0,0 +1,242 @@
+
+After the OIDC middleware signs the user in, the user is also automatically signed in to the cookie middleware. This allows the user to be authenticated on subsequent requests.
+
+For the `Logout` action, you need to sign the user out of both the Auth0 middleware as well as the cookie middleware.
+
+The `RedirectUri` passed in both instances indicates where the user is redirected after they log in or fail to log in.
+
+```cs
+// Controllers/AccountController.cs
+
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Auth0.AspNetCore.Mvc;
+
+public class AccountController : Controller
+{
+    public async Task Login(string returnUrl = "/")
+    {
+        var authenticationProperties = new AuthenticationPropertiesBuilder()
+            .WithRedirectUri(returnUrl)
+            .Build();
+
+        await HttpContext.ChallengeAsync(Constants.AuthenticationScheme, authenticationProperties);
+    }
+
+    [Authorize]
+    public async Task Logout()
+    {
+        var authenticationProperties = new AuthenticationPropertiesBuilder()
+            // Indicate here where Auth0 should redirect the user after a logout.
+            // Note that the resulting absolute Uri must be added to the
+            // **Allowed Logout URLs** settings for the app.
+            .WithRedirectUri(Url.Action("Index", "Home"))
+            .Build();
+
+        await HttpContext.SignOutAsync(Constants.AuthenticationScheme, authenticationProperties);
+        await HttpContext.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
+    }
+}
+```
+
+### Add the Login and Logout buttons
+
+Add the **Log In** and **Log Out** buttons to the navigation bar. In the `/Views/Shared/_Layout.cshtml` file, in the navigation bar section, add code that displays the **Log Out** button when the user is authenticated and the **Log In** button if not. The buttons link to the `Logout` and `Login` actions in the `AccountController`:
+
+```html
+<!-- Views/Shared/_Layout.cshtml -->
+<div class="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse">
+    <ul class="nav navbar-nav">
+        <li><a asp-area="" asp-controller="Home" asp-action="Index">Home</a></li>
+    </ul>
+    <ul class="nav navbar-nav navbar-right">
+        @if (User.Identity.IsAuthenticated)
+        {
+            <li><a id="qsLogoutBtn" asp-controller="Account" asp-action="Logout">Logout</a></li>
+        }
+        else
+        {
+            <li><a id="qsLoginBtn" asp-controller="Account" asp-action="Login">Login</a></li>
+        }
+    </ul>
+</div>
+```
+
+### Run the application
+
+When the user selects the **Log In** button, the SDK redirects them to the [Universal Login Page](/hosted-pages/login) in your Auth0 domain.
+
+## Obtain an Access Token for Calling an API
```

Comment:
Does it make sense to have this step in the 01-Login tutorial step? Generally our top steps are
1. Configure Auth0
2. Install/Configure SDK
3. Add Login
4. Add Logout
5. Show User Profile
6. What's next (link to more resources)